### PR TITLE
Upgrade Hive version

### PR DESCRIPTION
**Pull request by chenz4027**
https://issues.redhat.com/browse/ACM-3395

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -122,7 +122,7 @@ replace (
 	github.com/openshift/hive/apis => github.com/openshift/hive/apis v0.0.0-20221110093336-443b7d70c311
 	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.24.3
 	k8s.io/cli-runtime => k8s.io/cli-runtime v0.24.3
-	k8s.io/client-go => k8s.io/client-go v0.24.11
+	k8s.io/client-go => k8s.io/client-go v0.26.2
```
Comment (mikeshng) [inline go.mod:125]: Just double checking the k8s.io client-go is at version v0.26.2 but all the other k8s.io stay at the old k8s version v0.24.3. Is this intentional?

Comment (chenz4027) [reply go.mod:125]: Yeah it requires higher version client go, had this issue: 
```
GOFLAGS="" go build -o build/_output/manager-clusterclaims ./cmd/clusterclaims/main.go
# k8s.io/client-go/tools/record
vendor/k8s.io/client-go/tools/record/event.go:301:13: assignment mismatch: 1 variable but e.Watch returns 2 values
vendor/k8s.io/client-go/tools/record/event.go:349:13: assignment mismatch: 1 variable but recorder.ActionOrDrop returns 2 values
make: *** [Makefile.prow:13: compile] Error 2
```

Comment (chenz4027) [reply go.mod:125]: So does: https://github.com/stolostron/multicloud-operators-foundation/pull/596/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R35

Comment (mikeshng) [reply go.mod:125]: I think this PR should move up all the other k8s imports and not just client-go just as you link in the repo:
https://github.com/stolostron/multicloud-operators-foundation/blob/main/go.mod#L28-L40

**Pull request merged**